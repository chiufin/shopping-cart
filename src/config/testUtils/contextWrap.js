import { BrowserRouter } from 'react-router-dom'
import { shallow, mount } from 'enzyme'
import { shape } from 'prop-types'

const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
}

const createContext = () => ({
  context: { router },
  childContextTypes: {
    router: shape({}),
  },
})

// used for testing router wrapped components when content is need to Mount render
export function mountWrap(node) {
  return mount(node, createContext())
}

// used for testing router wrapped components when content is need to Shallow render
export function shallowWrap(node) {
  return shallow(node, createContext())
}
