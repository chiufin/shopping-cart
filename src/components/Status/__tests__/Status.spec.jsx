import React from 'react'
import { mountWrap, shallowWrap } from '../../../config/testUtils/contextWrap'
import Status from '../Status'

const mockStatusProps = {
  code: 404,
  children: '<div />',
}

const setup = (props, renderer = shallowWrap) => {
  const component = renderer(<Status {...props} />)
  return {
    getInstance: () => component,
  }
}

describe('Status', () => {
  it('should render', () => {
    const component = setup()
    expect(component).toBeDefined()
    expect(component).toMatchSnapshot()
  })

  it('should mount content properly', () => {
    const component = mountWrap(<Status {...mockStatusProps} />)
    expect(component).toBeDefined()
    expect(component).toMatchSnapshot()
  })
})
