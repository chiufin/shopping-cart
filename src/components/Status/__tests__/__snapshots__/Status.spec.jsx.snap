// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status should mount content properly 1`] = `
<Status
  code={404}
>
  <Route
    render={[Function]}
  >
    &lt;div /&gt;
  </Route>
</Status>
`;

exports[`Status should render 1`] = `
Object {
  "getInstance": [Function],
}
`;
